MaxText/layers/attentions.py

```diff
@@ -303,7 +304,12 @@ def compute_local_attention(self, attn_weights: Array, value: Array) -> tuple[Ar
     local_max = jnp.reshape(local_max, (local_max.shape[0], local_max.shape[1], local_max.shape[2] * local_max.shape[3], 1))
     local_sum = jnp.reshape(local_sum, (local_sum.shape[0], local_sum.shape[1], local_sum.shape[2] * local_sum.shape[3], 1))
 
-    local_out = self.wv_product(local_exps, value)
+    local_out = self.wv_product(local_exps, value, q_seq_len)
+
+    if self.reshape_q and q_seq_len == 1:
```

Comment:
Are there any cases where q_seq_len == 1 and we don't want to perform this reshape? If not we could remove the config option and only check for q_seq_len == 1?

Reply:
I haven not verified MQA/GQA so I am inclined to play safe here.

Reply:
+1, not I'm curious how the broadcasted dimensions work if n_kv != n and t != s.

Reply:
If your goal is to apply this to decode-phase only, should we use that condition instead of `q_seq_len == 1`?

Reply:
If the prefill phase has q_seq_len == 1 in dot_general, it will work for it too